include/gatekeeper_gk.h

```diff
@@ -118,10 +151,24 @@ struct gk_fib {
 
 			/*
 			 * Cache the whole Ethernet header when the @next_fib
-			 * action is GK_FWD_NEIGHBOR.
+			 * action is GK_FWD_NEIGHBOR_*_NET.
 			 */
-			struct ether_cache eth_cache;
+			struct ether_cache *eth_cache;
 		} grantor;
+
+		/*
+		 * When the action is GK_FWD_NEIGHBOR_*_NET, it stores all
+		 * the neighbors' Ethernet headers in a hash table.
+		 * The entries can be accessed according to its IP address.
+		 */
+		struct {
+			/*
+			 * Since IPv4 entries can be accessed easily, we store
+			 * the IPv4 and IPv6 entries separately.
+			 */
+			struct neighbor_hash_table neigh_tbl;
+			struct neighbor_hash_table neigh_tbl6;
```

Comment:
Given that each field `neigh` and `neigh6` are just `struct neighbor_hash_table`, you can declare them as follows:
```
struct neighbor_hash_table neigh;
struct neighbor_hash_table neigh6;
```